crates/prices/src/core/price.rs

```diff
@@ -0,0 +1,35 @@
+use {
+    super::{eth, swap},
+    std::ops::Mul,
+};
+
+/// The price between [`swap::FromToken`] and [`swap::ToToken`].
+#[derive(Debug, Clone, Copy)]
+pub struct Price {
+    from: swap::FromAmount,
+    to: swap::ToAmount,
+}
+
+impl Price {
+    pub fn new(from: swap::FromAmount, to: swap::ToAmount) -> Self {
+        Self { from, to }
+    }
+}
+
+/// Apply the price to the [`swap::FromAmount`].
+impl Mul<swap::FromAmount> for Price {
```

Comment:
Not sure if overloading the `*` operator is a great idea.
And if we do we `Output` should be `Result` because multiplications and divisions can panic which needs to be handled in some way.